carma_cooperative_perception/src/external_object_list_to_sdsm_component.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright 2023 Leidos
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "carma_cooperative_perception/external_object_list_to_sdsm_component.hpp"
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+
+#include "carma_cooperative_perception/msg_conversion.hpp"
+
+namespace carma_cooperative_perception
+{
+ExternalObjectListToSdsmNode::ExternalObjectListToSdsmNode(const rclcpp::NodeOptions & options)
+: CarmaLifecycleNode{options}
+{
```

Comment:
The `sdsm_publisher_` should be added to the base class's `lifecycle_publishers_` list so it can manage the publisher's lifecycle calls. See the `CarmaLifecycleNode` [header](https://github.com/usdot-fhwa-stol/carma-utils/blob/develop/carma_ros2_utils/include/carma_ros2_utils/carma_lifecycle_node.hpp) for details.